File: AngularFrontend/src/app/core/domain-classes/documentVersion.ts
export interface DocumentVersion {
  id?: string;
  documentId?: string;
  url?: string;
  CreatedByUser?: string;
  isCurrentVersion?: boolean;
  file?: string | Blob;
  isFileChange?: boolean;
  isSignatureExists?: boolean;
  signBy?: string;
  signDate?: Date;
  extension?: string;
  comment?: string;
  versionNumber?: number;
  isChunk?: boolean;
  modifiedDate?: Date;
  modifiedBy?: string;
  createdByUser?: string;
  createdDate?: Date;

}

File: MySQL-NET-REST-API/DocumentManagement.Data/Entities/DocumentVersion.cs
using DocumentManagement.Data.Entities;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentManagement.Data
{
    public class DocumentVersion : BaseEntity
    {
        public Guid Id { get; set; }
        public Guid DocumentId { get; set; }
        public string Url { get; set; }
        [ForeignKey("DocumentId")]
        public Document Document { get; set; }
        [ForeignKey("CreatedBy")]
        public User CreatedByUser { get; set; }
        [Column(TypeName = "LONGBLOB")]
        public byte[] Key { get; set; }
        [Column(TypeName = "LONGBLOB")]
        public byte[] IV { get; set; }
        public Guid? SignById { get; set; }
        [ForeignKey("SignById")]
        public User SignBy { get; set; }
        public DateTime? SignDate { get; set; }
        public string Comment { get; set; }
        public bool IsCurrentVersion { get; set; }
        public int VersionNumber { get; set; }
        public string Extension { get; set; }
        public bool IsChunk { get; set; }
        public bool IsAllChunkUploaded { get; set; }
        public ICollection<DocumentChunk> DocumentChunks { get; set; }

    }
}
